contracts/ZapLiquidity.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.15;
+
+/*
+  ______                     ______                                 
+ /      \                   /      \                                
+|  ▓▓▓▓▓▓\ ______   ______ |  ▓▓▓▓▓▓\__   __   __  ______   ______  
+| ▓▓__| ▓▓/      \ /      \| ▓▓___\▓▓  \ |  \ |  \|      \ /      \ 
+| ▓▓    ▓▓  ▓▓▓▓▓▓\  ▓▓▓▓▓▓\\▓▓    \| ▓▓ | ▓▓ | ▓▓ \▓▓▓▓▓▓\  ▓▓▓▓▓▓\
+| ▓▓▓▓▓▓▓▓ ▓▓  | ▓▓ ▓▓    ▓▓_\▓▓▓▓▓▓\ ▓▓ | ▓▓ | ▓▓/      ▓▓ ▓▓  | ▓▓
+| ▓▓  | ▓▓ ▓▓__/ ▓▓ ▓▓▓▓▓▓▓▓  \__| ▓▓ ▓▓_/ ▓▓_/ ▓▓  ▓▓▓▓▓▓▓ ▓▓__/ ▓▓
+| ▓▓  | ▓▓ ▓▓    ▓▓\▓▓     \\▓▓    ▓▓\▓▓   ▓▓   ▓▓\▓▓    ▓▓ ▓▓    ▓▓
+ \▓▓   \▓▓ ▓▓▓▓▓▓▓  \▓▓▓▓▓▓▓ \▓▓▓▓▓▓  \▓▓▓▓▓\▓▓▓▓  \▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓ 
+         | ▓▓                                             | ▓▓      
+         | ▓▓                                             | ▓▓      
+          \▓▓                                              \▓▓         
+
+ * App:             https://apeswap.finance
+ * Medium:          https://ape-swap.medium.com
+ * Twitter:         https://twitter.com/ape_swap
+ * Discord:         https://discord.com/invite/apeswap
+ * Telegram:        https://t.me/ape_swap
+ * Announcements:   https://t.me/ape_swap_news
+ * GitHub:          https://github.com/ApeSwapFinance 
+ */
+
+import "./interfaces/IZapLiquidity.sol";
+import "./interfaces/IArrakisRouter.sol";
+import "./interfaces/IArrakisPool.sol";
+import "./interfaces/INonfungiblePositionManager.sol";
+import "./interfaces/IApeFactory.sol";
+import "./interfaces/IApePair.sol";
+import "./libraries/Constants.sol";
+import "./libraries/MathHelper.sol";
+import "./utils/TransferHelper.sol";
+
+import "hardhat/console.sol";
+
+contract ZapLiquidity is IZapLiquidity, TransferHelper {
+    using SafeERC20 for IERC20;
+
+    function addLiquidityV2(
+        AddLiquidityV2Params memory params
+    ) external payable returns (uint256 amount0Lp, uint256 amount1Lp) {
```

Comment:
Because `multicall` and `wrapNative` are `payable` functions, it looks like this function doesn't need to have include `payable`. 

I do my best to remove `payable` where possible to prevent accidental native tranfers.

Reply:
We actually do need payable basically everywhere to allow native zap. Else it reverts

Reply:
Ok I see